### PR TITLE
fix: generating a payment request from a quotation instread of a sales order

### PR DESCRIPTION
**Pull request by rvpasari**

### CODE REVIEW

**Comment by neilLasrado** [comment]
@rvpasari Squash your commits - http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: erpnext/accounts/doctype/payment_request/payment_request.py [hunk 1]
```diff
@@ -164,15 +167,48 @@ def create_payment_entry(self, submit=True):
...
 		else:
 			party_amount = self.grand_total
 
-		payment_entry = get_payment_entry(self.reference_doctype, self.reference_name,
-			party_amount=party_amount, bank_account=self.payment_account, bank_amount=bank_amount)
+		#create a sales order if the payment request has been made against a quotation from a shopping cart
+		if self.reference_doctype == "Quotation" and (hasattr(ref_doc, "order_type") and getattr(ref_doc, "order_type") == "Shopping Cart"):
+
+			#we submit the quotation
+			frappe.db.set_value(self.reference_doctype, self.reference_name, "docstatus", 1)
```
Comment (neilLasrado) [inline erpnext/accounts/doctype/payment_request/payment_request.py]: do a get_doc and then do a submit

Comment (neilLasrado) [reply erpnext/accounts/doctype/payment_request/payment_request.py]: frappe.get_doc(self.reference_doctype, self.reference_name).submit()

File: erpnext/accounts/doctype/payment_request/payment_request.py [hunk 2]
```diff
@@ -164,15 +167,48 @@ def create_payment_entry(self, submit=True):
...
+			frappe.db.commit()
+
+			#convert the quotation to a sales order
+			from erpnext.selling.doctype.quotation.quotation import _make_sales_order
+			sales_order = frappe.get_doc(_make_sales_order(self.reference_name, ignore_permissions=True))
+			sales_order.payment_schedule = []
+			sales_order.flags.ignore_permissions = True
+			sales_order.insert()
+			sales_order.submit()
+			print(sales_order.as_dict())
```
Comment (neilLasrado) [inline erpnext/accounts/doctype/payment_request/payment_request.py]: remove the print

File: erpnext/shopping_cart/cart.py [hunk 3]
```diff
@@ -83,20 +83,13 @@ def place_order():
...
-
-	return sales_order.name
+	return quotation.name
 
 @frappe.whitelist()
 def request_for_quotation():
 	quotation = _get_cart_quotation()
 	quotation.flags.ignore_permissions = True
-	quotation.submit()
+	#quotation.submit()
```
Comment (neilLasrado) [inline erpnext/shopping_cart/cart.py]: ?

**Review by neilLasrado — CHANGES_REQUESTED** [review]

File: erpnext/shopping_cart/cart.py (hunk 3 above)
Comment (rvpasari) [reply erpnext/shopping_cart/cart.py]: commenting it out so that it doesn't get submitted when the function is called

Comment (neilLasrado) [reply erpnext/shopping_cart/cart.py]: remove the line insted

**Review by neilLasrado — APPROVED** [review]

**Review by DeveloperMetal — APPROVED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**